Build Page Tree: Print Conversion Failure Details

When building the page tree, if any of the AsciiDoc source files could not be processed due to an exception, please provide some details (like the page that failed).

Currently I am getting this exception trace. Notice the absence of an indicate of which page failed, and why:
```
Exception in thread "main" java.lang.RuntimeException: Could not convert asciidoc pages
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluenceConverter.convert(AsciidocConfluenceConverter.java:91)
	at org.sahli.asciidoc.confluence.publisher.cli.AsciidocConfluencePublisherCommandLineClient.main(AsciidocConfluencePublisherCommandLineClient.java:66)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1967)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.lambda$null$7(AsciidocConfluencePage.java:196)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.replaceAll(AsciidocConfluencePage.java:151)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.lambda$replaceCrossReferenceTargets$8(AsciidocConfluencePage.java:194)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.lambda$postProcessContent$4(AsciidocConfluencePage.java:161)
	at java.util.stream.ReduceOps$1ReducingSink.accept(ReduceOps.java:80)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:484)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.postProcessContent(AsciidocConfluencePage.java:161)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.convertedContent(AsciidocConfluencePage.java:122)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.newAsciidocConfluencePage(AsciidocConfluencePage.java:106)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluenceConverter.lambda$buildPageTree$0(AsciidocConfluenceConverter.java:103)
	at java.util.ArrayList.forEach(ArrayList.java:1255)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluenceConverter.buildPageTree(AsciidocConfluenceConverter.java:98)
	at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluenceConverter.convert(AsciidocConfluenceConverter.java:82)
	... 1 more
```

--- a/asciidoc-confluence-publisher-converter/src/main/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePage.java
+++ b/asciidoc-confluence-publisher-converter/src/main/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePage.java
@@ -131,8 +131,8 @@ public class AsciidocConfluencePage {
             List<String> keywords = keywords(asciidocContent);
 
             return new AsciidocConfluencePage(pageTitle, pageContent, attachmentCollector, keywords);
-        } catch (IOException e) {
-            throw new RuntimeException("Could not create asciidoc confluence page", e);
+        } catch (Exception e) {
+            throw new RuntimeException("failed to create confluence page for asciidoc content in '" + asciidocPage.path().toAbsolutePath() + " '", e);
         }
     }
 

--- a/asciidoc-confluence-publisher-converter/src/test/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePageTest.java
+++ b/asciidoc-confluence-publisher-converter/src/test/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePageTest.java
@@ -16,6 +16,8 @@
 
 package org.sahli.asciidoc.confluence.publisher.converter;
 
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
 import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
@@ -52,6 +54,7 @@ import static org.junit.rules.ExpectedException.none;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 import static org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.newAsciidocConfluencePage;
+import static org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePageTest.RootCauseMatcher.rootCauseWithMessage;
 
 /**
  * @author Alain Sahli
@@ -216,7 +219,8 @@ public class AsciidocConfluencePageTest {
 
         // assert
         this.expectedException.expect(RuntimeException.class);
-        this.expectedException.expectMessage("top-level heading or title meta information must be set");
+        this.expectedException.expectMessage("failed to create confluence page for asciidoc content in");
+        this.expectedException.expect(rootCauseWithMessage("top-level heading or title meta information must be set"));
 
         // act
         newAsciidocConfluencePage(asciidocPage(adoc), UTF_8, TEMPLATES_FOLDER, dummyAssetsTargetPath());
@@ -1703,6 +1707,31 @@ public class AsciidocConfluencePageTest {
         } catch (Exception e) {
             throw new RuntimeException("Could not set default charset", e);
         }
+    }
+
+
+    static class RootCauseMatcher extends TypeSafeMatcher<Exception> {
+
+        private final String message;
+
+        private RootCauseMatcher(String message) {
+            this.message = message;
+        }
+
+        @Override
+        protected boolean matchesSafely(Exception exception) {
+            return exception.getCause().getMessage().equals(this.message);
+        }
+
+        @Override
+        public void describeTo(Description description) {
+            description.appendText("root cause with message '" + this.message + "'");
+        }
+
+        static RootCauseMatcher rootCauseWithMessage(String message) {
+            return new RootCauseMatcher(message);
+        }
+
     }
 
 }

